agenta-cli/agenta/sdk/tracing/llm_tracing.py

```diff
@@ -322,22 +318,98 @@ def close_span(self, outputs: Dict[str, Any]) -> None:
             self._update_span_cost(tracing.active_span, outputs.get("cost", None))
             self._update_span_tokens(tracing.active_span, outputs.get("usage", None))
 
-        tracing.closed_spans.append(tracing.active_span)
-
         active_span_parent_id = tracing.active_span.parent_span_id
 
-        if active_span_parent_id is None:
-            self.end_trace(parent_span=tracing.active_span)
-
-        else:
-            parent_span = tracing.tracked_spans[active_span_parent_id]
+        if active_span_parent_id is not None:
+            parent_span = tracing.spans[active_span_parent_id]
             self._update_span_cost(parent_span, tracing.active_span.cost)
             self._update_span_tokens(parent_span, tracing.active_span.tokens)
             tracing.active_span = parent_span
         ### --- TO BE CLEANED --- <<<
 
         logging.info(f"Closed  span  {span_id} {spankind}")
 
+    def store_locals(self, locals: Dict[str, Any] = {}):
+        attributes = {f"locals.{k}": v for k, v in locals.items()}
+
+        self.set_attributes(attributes=attributes)
+
+    def dump_trace(self):
+        try:
```

Comment:
I'd suggest we add multi-line documentation to avoid re-explaining code that might come off as _advanced/complex_. 

```suggestion
    """Collects and organizes tracing information into a dictionary.

    This function retrieves the current tracing context and extracts relevant data such as `trace_id`, `cost`, `tokens`, and `latency` for the root span (the span without a `parent_span_id`). It also dumps detailed span information using the `dump_spans` method and includes it in the trace dictionary. If an error occurs during the process, it logs the error message and stack trace.

    Returns:
        dict: A dictionary containing the trace information.
    """
        try:
```

Reply:
Thank you :)

Reply:
;-)